src/server/connection.py

```diff
@@ -155,14 +183,37 @@ def close(self, connection: ChiaConnection, keep_peer: bool = False):
             connection.close()
             self._state_changed("close_connection")
             if not keep_peer:
-                self.peers.remove(info)
+                if self.introducer_peers is not None:
+                    self.introducer_peers.remove(info)
 
     def close_all_connections(self):
         for connection in self._all_connections:
             connection.close()
             self._state_changed("close_connection")
         self._all_connections = []
-        self.peers = Peers()
+        if self.local_type == NodeType.INTRODUCER:
+            self.introducer_peers = IntroducerPeers()
+
+    def get_local_peerinfo(self):
+        ip = None
+        port = None
+        for c in self._all_connections:
+            if c.connection_type == NodeType.FULL_NODE:
+                port = c.local_port
+                break
+        if port is None:
+            return (None, None)
```

Comment:
I think the right way to do this is to return Optional[Tuple[..]], and checking is None on the calling side, as opposed to returning (None, None).